lib/tutorials/caching.md

```diff
@@ -0,0 +1,187 @@
+## Client side caching
+
+Http protocol provides several mechanisms how to cache resources in browser via http headers. To decide which headers are suitable for your use case check Google developers [guide](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/http-caching) or other [resources](https://www.google.com/search?q=HTTP+caching). This tutorial provides overview how to use these techniques in hapi.
+
+### Cache-Control
+
+`Cache-Control` header tells browser and any intermediate cache if resource is cacheable and for how long. For example `Cache-Control:max-age=30, must-revalidate, private` means that browser can cache resource for 30 seconds and `private` means that it should not be cached by intermediate caches, only by browser. `must-revalidate` means that once it expires it has to request the resource again.
+
+Lets see how we can set this header in hapi:
+
+```javascript
+server.route({
+   path: '/hapi/{ttl?}',
+   method: 'GET',
+   handler: function (request, reply) {
+
+      var response = reply({be: 'hapi'});
+      if (request.params.ttl) {
+         response.ttl(request.params.ttl);
+      }
+   },
+   config: {
+      cache: {
+         expiresIn: 30 * 1000,
+         privacy: 'private'
+      }
+   }
+});
+```
+
+Example also illustrates that `expiresIn` value can be overridden with `ttl(msec)` function using [response object](http://hapijs.com/api#response-object) interface.
+
+See [route-options](http://hapijs.com/api#route-options) for more options.
+### Last-Modified
+
+In some cases server can provide information when resources was last modified. Good example is [inert](https://github.com/hapijs/inert) module, which is responsible for serving static files in hapi (`file` and `directory` handler). Last modification timestamp is read for every requested file and send as `Last-Modified` header.
+
+When `Last-Modified` is set on response hapi compares it with `If-Modified-Since` coming from client to decide if should return `304`.
+
+
+Assuming `lastModified` is Date object you can set this header via [response object](http://hapijs.com/api#response-object) interface.
```

Comment:
Do you mean whole Last-Modified section? There is good example how to use Last-Modified with server side caching that can be pretty useful (in last section).